ide/db.sql.editor/src/org/netbeans/modules/db/sql/editor/completion/SQLCompletionQuery.java

```diff
@@ -120,23 +120,37 @@ public void query(SQLCompletionResultSet resultSet, SQLCompletionEnv newEnv) {
 
     private void doQuery(final SQLCompletionEnv newEnv) {
         try {
+            // DB Connection available
+            if (dbconn != null) {
```

Comment:
Not sure I completely understand (trying to learn here).  What might cause that to be changed to null?  Is the suggestion in some other context (maybe another thread or file window), the dbconn is set (or unset) in some other way?  

The dbconn is set as part of the SQLCompletionQuery constructor which is used during the "SQLCompletionProvider.createTask" with a AsyncCompletionTask constructor, using a dbconn from that context.  I believe in those context it may be creating a new instance, but not 100% sure here.

Reply:
If dbconn can't be set, all is save - that is the answer. This work is all about not having a connection, so I think it is warranted to be save about this.